src/Tools/dotnet-counters/Exporters/ConsoleWriter.cs

```diff
@@ -430,10 +416,16 @@ public void Stop()
 
                     if (row > -1)
                     {
-                        SetCursorPosition(0, row);
-                        Console.WriteLine();
+                        _console.SetCursorPosition(0, row);
+                        _console.WriteLine();
                     }
                 }
+
+                try
+                {
+                    _console.CursorVisible = oldCursorVisibility;
```

Comment:
I don't see a CancelKeyPress handler, though I might have missed it.  If the user kills the process with Control-C, whether or not the cursor is properly restored or not will depend on the terminal implementation.  That means some Control-C behavior will leave the cursor invisible in some consoles, which is quite annoying.

I would suggest adding a CancelKeyPress handler to your IConsole interface if you are messing with cursor visibility and register it when you set IConsole.CursorVisible successfully.

Lastly, you might consider making `oldCursorVisibility` a nullable and check its nullability in Initialize, to make sure you haven't already set the cursor/ControlC handler.  It might be a bug if someone calls Initialize twice, but it doesn't hurt to get cursor visibility restoration right.

I ran into this a few times with an old project I had where hitting Control-C would sometimes mess up that terminal window's cursor, requiring a restart.

Reply:
thanks!

Reply:
I'll probably just take this out for now. Even with a ctrl-C handler there are other failure modes that wouldn't hit it and if the terminal won't restore it on process exit I don't want to mess up people's console just for a small aesthetic improvement.